retry-promise-on-rejection.js

```diff
@@ -0,0 +1,25 @@
+
+/**
+ * @param {() => Promise<any>} fetcher
+ * @param {number} maximumRetryCount
+ * @return {Promise<any>}
+ */
+
+function fetchWithAutoRetry(fetcher, maximumRetryCount) {
+  let numberOfErrors = 0;
+
+  const recursive = () => {
+    return fetcher()
+      .catch((err) => {
+        numberOfErrors++;
+
+        if (numberOfErrors <= maximumRetryCount) {
+          return recursive();
+        }
+
+        throw err;
+      });
+  }
+
+  return recursive();
+}
```

Comment:
you solved it perfectly through recursion
also this problem can be solved by iterating